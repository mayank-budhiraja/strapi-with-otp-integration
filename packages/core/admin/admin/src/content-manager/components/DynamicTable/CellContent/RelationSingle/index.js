import React from 'react';
import PropTypes from 'prop-types';
import { Typography } from '@strapi/design-system/Typography';
import styled from 'styled-components';
import CellValue from '../CellValue';

const TypographyMaxWidth = styled(Typography)`
  max-width: 500px;
`;

const RelationSingle = ({ metadatas, value }) => {
  return (
    <TypographyMaxWidth textColor="neutral800" ellipsis>
      <CellValue
        type={metadatas.mainField.schema.type}
        value={value[metadatas.mainField.name] || value.id}
      />
    </TypographyMaxWidth>
  );
};

RelationSingle.propTypes = {
  metadatas: PropTypes.shape({
    mainField: PropTypes.shape({
      name: PropTypes.string.isRequired,
      schema: PropTypes.shape({ type: PropTypes.string.isRequired }).isRequired,
    }),
  }).isRequired,
  value: PropTypes.object.isRequired,
};

export default RelationSingle;
