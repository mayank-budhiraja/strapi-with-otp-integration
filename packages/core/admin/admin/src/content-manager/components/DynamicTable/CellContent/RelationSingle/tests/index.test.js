import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider, lightTheme } from '@strapi/design-system';
import { IntlProvider } from 'react-intl';

import RelationSingle from '../index';

const DEFAULT_PROPS_FIXTURE = {
  metadatas: {
    mainField: {
      name: 'name',
      schema: {
        type: 'string',
      },
    },
  },
  value: {
    count: 1,
  },
};

const ComponentFixture = () => {
  return (
    <ThemeProvider theme={lightTheme}>
      <IntlProvider locale="en" messages={{}} defaultLocale="en">
        <RelationSingle {...DEFAULT_PROPS_FIXTURE} />
      </IntlProvider>
    </ThemeProvider>
  );
};

describe('DynamicTabe / Cellcontent / RelationSingle', () => {
  it('renders and matches the snapshot', async () => {
    const { container } = render(<ComponentFixture />);
    expect(container).toMatchSnapshot();
  });
});
