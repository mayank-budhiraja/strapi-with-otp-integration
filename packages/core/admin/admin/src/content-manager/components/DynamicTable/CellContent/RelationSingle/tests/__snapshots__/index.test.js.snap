// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicTabe / Cellcontent / RelationSingle renders and matches the snapshot 1`] = `
.c2 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c0 {
  color: #32324d;
  display: block;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-size: 0.875rem;
  line-height: 1.43;
}

.c1 {
  max-width: 500px;
}

<div>
  <span
    class="c0 c1"
  >
    
  </span>
  <div
    class="c2"
  >
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-log"
      role="log"
    />
    <p
      aria-live="polite"
      aria-relevant="all"
      id="live-region-status"
      role="status"
    />
    <p
      aria-live="assertive"
      aria-relevant="all"
      id="live-region-alert"
      role="alert"
    />
  </div>
</div>
`;
